File: frontend/lib/api/favorites.ts
import { request } from "./client";
import type {
  FavoriteFolder,
  FavoriteVideosResponse,
  OrganizePreviewResponse,
} from "./sourceTypes";

export const favoritesApi = {
  getList: (sessionId: string) =>
    request<FavoriteFolder[]>("/favorites/list", {
      query: { session_id: sessionId },
    }),

  getAllVideos: (mediaId: number, sessionId: string) =>
    request<FavoriteVideosResponse>(`/favorites/${mediaId}/all-videos`, {
      query: { session_id: sessionId },
    }),

  organizePreview: (folderId: number, sessionId: string) =>
    request<OrganizePreviewResponse>("/favorites/organize/preview", {
      method: "POST",
      query: { session_id: sessionId },
      body: JSON.stringify({ folder_id: folderId }),
    }),

  organizeExecute: (
    data: {
      default_folder_id: number;
      moves: Array<{
        resource_id: number;
        resource_type: number;
        target_folder_id: number;
      }>;
    },
    sessionId: string,
  ) =>
    request<{ message: string; moved: number; groups: number }>(
      "/favorites/organize/execute",
      {
        method: "POST",
        query: { session_id: sessionId },
        body: JSON.stringify(data),
      },
    ),

  cleanInvalid: (folderId: number, sessionId: string) =>
    request<{ message: string; data: unknown }>(
      "/favorites/organize/clean-invalid",
      {
        method: "POST",
        query: { session_id: sessionId },
        body: JSON.stringify({ folder_id: folderId }),
      },
    ),
};

File: app/routers/favorites.py
"""
Bilibili RAG 知识库系统

收藏夹路由 - 旧 session_id 驱动接口已禁用，仅保留 410 Gone 提示
"""

from fastapi import APIRouter

from app.services.legacy_api import (
    LEGACY_FAVORITES_API_DETAIL,
    raise_legacy_api_gone,
)

router = APIRouter(prefix="/favorites", tags=["收藏夹"])


@router.get("/list")
async def get_favorites_list():
    """获取用户的收藏夹列表（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)


@router.get("/{media_id}/videos")
async def get_favorite_videos(media_id: int):
    """获取收藏夹中的视频列表（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)


@router.get("/{media_id}/all-videos")
async def get_all_favorite_videos(media_id: int):
    """获取收藏夹中的所有视频（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)


@router.post("/organize/preview")
async def organize_preview():
    """收藏夹整理预览（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)


@router.post("/organize/execute")
async def organize_execute():
    """执行收藏夹整理（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)


@router.post("/organize/clean-invalid")
async def clean_invalid_resources():
    """清理收藏夹失效内容（已禁用）"""
    raise_legacy_api_gone(LEGACY_FAVORITES_API_DETAIL)
